src/wagtail_personalisation/adapters.py

```diff
@@ -137,7 +137,7 @@ def add_page_visit(self, page):
             visit_count.append({
                 'slug': page.slug,
                 'id': page.pk,
-                'path': self.request.path,
+                'path': path or self.request.path,
```

Comment:
I'm now actually wondering if using self.request.path for fallback here is the correct idea. Perhaps we should always use the full path based on the passed page ?